internal/util/util_test.go

```diff
@@ -0,0 +1,154 @@
+package util
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	plain "github.com/operator-framework/rukpak/internal/provisioner/plain/types"
+)
+
+func TestCheckDesiredBundleTemplate(t *testing.T) {
+	sampleSpec := v1alpha1.BundleSpec{
+		ProvisionerClassName: plain.ProvisionerID,
+		Source: rukpakv1alpha1.BundleSource{
+			Type: rukpakv1alpha1.SourceTypeImage,
+			Image: &rukpakv1alpha1.ImageSource{
+				Ref: "non-existent",
+			},
+		},
+	}
+	type args struct {
+		existingBundle *rukpakv1alpha1.Bundle
+		desiredBundle  *rukpakv1alpha1.BundleTemplate
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "True/BundleMatchesTemplate",
+			args: args{
+				existingBundle: &rukpakv1alpha1.Bundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "stub",
+						Labels: map[string]string{
+							"stub": "stub",
+						},
+					},
+					Spec: sampleSpec,
+				},
+				desiredBundle: &rukpakv1alpha1.BundleTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "stub",
+						Labels: map[string]string{
+							"stub": "stub",
+						},
+					},
+					Spec: sampleSpec,
+				},
+			},
+			want: true,
+		},
+		{
+			name: "False/SpecDiffers",
+			args: args{
+				existingBundle: &rukpakv1alpha1.Bundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "stub",
+						Labels: map[string]string{
+							"stub": "stub",
+						},
+					},
+					Spec: rukpakv1alpha1.BundleSpec{
+						ProvisionerClassName: "non-existent-provisioner-class-name",
+					},
```

Comment:
nit: This is an invalid bundle.  It may be better adding the `Source` as-is here.

Reply:
Yep - I'll probably hold off on addressing this though as I think I'll refactor this test to use gomega as a follow-up.